torch/_inductor/triton_heuristics.py

```diff
@@ -173,6 +173,7 @@ def _precompile_config(self, cfg: Config, warm_cache_only_with_cc: int):
         compile_meta["debug"] = (
             config.triton.assert_indirect_indexing and torch.version.hip is None
         )
+        compile_meta["device_type"] = "cuda" if torch.version.hip is None else "hip"
```

Comment:
@jataylo We should add a comment in the code explaining why this is needed, otherwise this will seem like "voodoo magic" sometime down the line.